feat: add `make:suite` command

Julien-R44 opened the pull request:
Based on #47 
Review from : fcb94d214fd32806313e10fabe6387ab330268c8

Sorry for the git mess ! I should have made my japa-migration branch on adonisjs/assemble and made a PR on it
Since japa-migration is on my fork, I don't think there is any other solution. I hope it won't cause too much problem for merge but normally it should be ok 😓

---

So I added the `node ace create-test-suite` command !
`node ace create-test-suite my-suite` add this to the RC File : 
```json
{
  "name": "my-suite",
  "files": [ "tests/my-suite/**/*.spec(.ts|.js)" ],
  "timeout": 60000
},
````

And create a sample file at "./tests/my-suite/test.spec.ts" unless the argument `--add-sample-test-file` is explicitly disabled.

I had a doubt about naming it `make:suite`, because I didn't originally plan to create a test sample file with the suite, so I thought it was weird to name it that way knowing that this command didn't generate anything and only modified the rc file.

But even now I'm not sure if this name is appropriate. Let me know what you think about that

thetutlage commented on commands/CreateSuite.ts:
```diff
@@ -0,0 +1,91 @@
+   * Name of the test suite to be created
+   */
+  @args.string({ description: 'Name of the test suite' })
+  public suiteName: string
```
The property name can be just `name`. Because the class makes it clear that we are dealing with a suite

thetutlage commented on commands/CreateSuite.ts:
```diff
@@ -0,0 +1,91 @@
+   * Should add a sample test file
+   */
+  @flags.boolean({ description: 'Add a sample test file' })
+  public addSampleTestFile: boolean = true
```
This can renamed to `withExampleTest` and can be mentioned as `--with-example-test`

thetutlage reviewed (changes requested):
We can also prompt the user to define the path for the test files. Something like

```
Enter path to the test files for this suite
```

The same can be defined as a flag as well

```
node ace make:suite unit --location="tests/unit"
```

We might have to check the path for `glob` pattern or the file extensions and if both are missing then add the default glob.

```ts
if (!['*', '.js', '.ts'].find((keyword) => this.location.includes(keyword)) {
  this.location = `${this.location}/**/*.spec(.ts|.js)`
}
```

Julien-R44 commented:
Are you sure about the prompt? None of our `make` commands ask the user for anything, everything goes through arguments.

thetutlage commented:
> Are you sure about the prompt? None of our make commands ask the user for anything, everything goes through arguments.

True. Let's make it configure via flag and skip the prompt. Infact, we can make the `location` an argument.

```sh
node ace make:suit unit tests/unit/**.spec.ts --with-example-test
```

The `location` argument is going to be optional

Julien-R44 commented:
We should be pretty good now !

- I renamed the command to `make:suite`
- Changed arguments/flags names
- `--location` can now accept a path or a glob pattern directly

See : https://github.com/adonisjs/assembler/pull/48/files#diff-b594ffbf42007fdfb2261a33aefc3d65d3a6907a2ac292566e385b6995e08773R90

To see how the final glob pattern is generated. 
I basically followed what you proposed : 
- Default glob pattern is used when the "location" does not include a wildcard or ends with an extension. 
- If it's just a folder, then we concatenate the end of the globPattern, so that gives : `/tests/${suiteName}/**/*.spec.(.ts|.js)`
- If it's a already a glob pattern, we simply add it directly to the Rc file and we get the parent folder with `globParent` to add the sample file

thetutlage commented on commands/Make/Suite.ts:
```diff
@@ -0,0 +1,120 @@
+
+import { BaseCommand, args, flags } from '@adonisjs/core/build/standalone'
+import { files, logger } from '@adonisjs/sink'
+import globParent from 'glob-parent'
```
I cannot see `glob-parent` as a dependency inside `package.json` file. Have I missed something?

Julien-R44 replied:
Yep 😋
https://github.com/adonisjs/assembler/blob/develop/package.json#L108

Already on the develop branch and used by the make:test command

thetutlage replied:
Ahh okay

thetutlage commented:
Looks good. Thanks for the PR :)
